docs: fix changelog (#570)
scikit-build/scikit-build: CHANGES.rst

Scikit-build 0.12.0
===================

> The scikit-build GitHub organization welcomes :user:`henryiii` and :user:`mayeut` as core contributors
> and maintainers.

Reviewer: ```suggestion
The scikit-build GitHub organization welcomes :user:`henryiii` and :user:`mayeut` as core contributors
and maintainers. Both are also maintainers of `cibuildwheel <https://cibuildwheel.readthedocs.io>`_.
```

> :user:`henryiii` has been instrumental to add Apple Silicon support, add support for Visual Studio 2019, update
> the Continuous Integration infrastructure as well as helping review & integrate contributions, and address
> miscellaneous issues. Additionally, :user:`henryiii` worked on adding `example project <https://github.com/pybind/scikit_build_example>`_
> to built with ``pybind11`` and ``scikit-build``.

Reviewer: ```suggestion
:user:`henryiii` is a `pybind11 <https://pybind11.readthedocs.io>`_ and `pypa/build <https://pypa-build.readthedocs.io>`_ maintainer, has been instrumental in adding Apple Silicon support, adding support for Visual Studio 2019, updating
the Continuous Integration infrastructure, as well as helping review & integrate contributions, and addressing
miscellaneous issues. Additionally, :user:`henryiii` has worked on an `example project <https://github.com/pybind/scikit_build_example>`_  to build with ``pybind11`` and ``scikit-build``.
```

> :user:`mayeut` is the maintainer of `cibuildwheel <https://cibuildwheel.readthedocs.io>`_ and
> focused his effort on updating the ``cmake-python-distributions`` and ``ninja-python-distributions`` so
> that the corresponding wheels are available on all supported platforms including Apple Silicon and all flavors
> of manylinux.
> 

Reviewer: ```suggestion
:user:`mayeut` is a `manylinux <https://github.com/pypa/manylinux>`_ maintainer and
focused his effort on updating the ``cmake-python-distributions`` and ``ninja-python-distributions`` so
that the corresponding wheels are available on all supported platforms including Apple Silicon and all flavors
of manylinux.

```

@mayeut is out for a couple of weeks, but I think this is a bit better, if you want to list related projects we are maintainers of.
